src/app/revenue-share/revenue-share.component.ts

```diff
@@ -20,6 +37,63 @@ export class RevenueShareComponent implements OnInit {
 
     fetchDataFromRoute() {
         this.projectID = this.route.snapshot.params.projectID;
-        this.amount = this.route.snapshot.params.amount;
+        this.amountInvested = this.route.snapshot.params.amount;
+        this.getProject(this.projectID);
+    }
+
+    getProject(projectUUID: string) {
+        SpinnerUtil.showSpinner();
+        this.projectService.getProject(projectUUID)
+            .subscribe(res => {
+                SpinnerUtil.hideSpinner();
+                this.projectName = res.name;
+            }, error => {
+                hideSpinnerAndDisplayError(error);
+            });
+    }
+
+    generateTransactionForRevenuePayout(amountInvested: number) {
+        SpinnerUtil.showSpinner();
+        this.revenueShareService.generateRevenueShareTx(this.projectID, amountInvested)
+            .subscribe(async (res) => {
+                const arkaneConnect = new ArkaneConnect('AMPnet', {environment: 'staging'});
+                const acc = await arkaneConnect.flows.getAccount(SecretType.AETERNITY);
+                const sigRes = await arkaneConnect.createSigner(WindowMode.POPUP).sign({
+                    walletId: acc.wallets[0].id,
+                    data: res.tx,
+                    type: SignatureRequestType.AETERNITY_RAW
+                });
+                this.broadcastService.broadcastSignedTx(sigRes.result.signedTransaction, res.tx_id)
+                    .subscribe(_ => {
+                        swal('', 'Successful revenue payout!', 'success');
+                        SpinnerUtil.hideSpinner();
+                    }, err => {
+                        hideSpinnerAndDisplayError(err);
+                    });
+            }, err => {
+                hideSpinnerAndDisplayError(err);
+            });
+    }
+
+    showRevenueConfirmModal() {
+        this.modalRef = this.modalService.show(RevenueShareConfirmModalComponent);
+        this.modalRef.content.onConfirmClicked.subscribe(amount => {
+            this.amountInvestedConfirm = amount;
+            this.checkInvestedConfirmedAmount();
+        });
+    }
+
+    checkInvestedConfirmedAmount() {
```

Comment:
The name of this function suggest just checking the confirmed amount, but it also generates a transaction which is breaking a [single responsibility principle](https://en.wikipedia.org/wiki/Single-responsibility_principle) of that function. You should rename the function and make the code more readable.